### PR TITLE
Support route on ios

### LINKED ISSUE #4703: Support `--route` on iOS
I have not tried on iOS Device.
Works on an Android device (--debug mode, did not try emulator or other modes).

```
cd examples/flutter_gallery
flutter run --route /buttons
```

Should load up to the "Buttons" demo (lib/demo/buttons_demo.dart), but just takes you to the Gallery main page instead.

### PR DESCRIPTION
**Pull request by Jasguerrero**
#4703

This depend on an engine PR to merge first: https://github.com/flutter/engine/pull/31555

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -3289,6 +3289,16 @@ targets:
       task_name: external_ui_integration_test_ios
     scheduler: luci
 
+  - name: Mac_ios route_test
+    recipe: devicelab/devicelab_drone
+    presubmit: false
```
Comment (christopherfujino) [inline .ci.yaml:3294]: this should have `bringup: true`

File: TESTOWNERS [hunk 2]
```diff
@@ -61,7 +61,7 @@
 /dev/devicelab/bin/tasks/platform_channels_benchmarks.dart @gaaclarke @flutter/engine
 /dev/devicelab/bin/tasks/platform_views_scroll_perf__timeline_summary.dart @zanderso @flutter/engine
 /dev/devicelab/bin/tasks/plugin_dependencies_test.dart @jmagman @flutter/tool
-/dev/devicelab/bin/tasks/routing_test.dart @zanderso @flutter/tool
+/dev/devicelab/bin/tasks/route_test.dart @zanderso @flutter/tool
```
Comment (Jasguerrero) [inline TESTOWNERS:64]: Does the renaming of this test file affect metrics? @keyonghan

Comment (christopherfujino) [reply TESTOWNERS:64]: I think it's renaming the builder in the .ci.yaml that might be an issue: we're already running this as "Linux_android routing_test" today: https://ci.chromium.org/p/flutter/builders/prod/Linux_android%20routing_test.

Comment (christopherfujino) [reply TESTOWNERS:64]: Oh, I see. @Jasguerrero do we need to change the task name?

Comment (Jasguerrero) [reply TESTOWNERS:64]: @christopherfujino  I don't think is necessary if it brings more problems than what it solves (just kind of confusing the task name is routing_test and the builder is route_test). I will be renaming it back to "routing_test"

**Review by CaseyHillers — APPROVED** [review]

File: TESTOWNERS [hunk 3]
```diff
@@ -170,6 +170,7 @@
 /dev/devicelab/bin/tasks/post_backdrop_filter_perf_ios__timeline_summary.dart @zanderso @flutter/engine
 /dev/devicelab/bin/tasks/simple_animation_perf_ios.dart @zanderso @flutter/engine
 /dev/devicelab/bin/tasks/tiles_scroll_perf_ios__timeline_summary.dart @zanderso @flutter/engine
+/dev/devicelab/bin/tasks/routing_test.dart @zanderso @flutter/tool
```
Comment (keyonghan) [inline TESTOWNERS:173]: This entry is not needed as we have already defined it in the linux section. 
Could you please add a section like `## Linux android and Mac iOS Devicelab tests` and add the shared entry there?

**Comment by christopherfujino** [comment]
Looks like the engine change for this should be in the framework: https://github.com/flutter/flutter/pull/99446

**Review by keyonghan — APPROVED** [review]
Thanks for the update! LGTM.

**Pull request merged**